### PR TITLE
Updated trust

### PR DESCRIPTION
**Pull request by akowal3**
yes.

### CODE REVIEW

File: internal/clients/team4/client.go [hunk 1]
```diff
@@ -21,14 +21,18 @@ func init() {
...
-		riskTaking:       0,
-		maxPardonTime:    3,
-		maxTierToPardon:  shared.SanctionTier3,
-		minTrustToPardon: 0.6,
+		greediness:                   0,
+		selfishness:                  0,
+		fairness:                     0,
+		collaboration:                0,
+		riskTaking:                   0,
+		maxPardonTime:                3,
```
Comment (tobhil98) [inline internal/clients/team4/client.go:29]: Just check with @MikeIsMike if he knows if there are any issues when initialising to 0 and not 0.5 for example

File: internal/clients/team4/client.go [hunk 2]
```diff
@@ -272,7 +285,47 @@ func (c *client) StartOfTurn() {
...
+		}
+		c.savedHistory.updated = false
+	}
+}
+
+func (c *client) updateTrustMonitoring(data map[shared.CommunicationFieldName]shared.CommunicationContent) {
+	if roleMonitored, ok := data[shared.RoleMonitored]; ok {
+		if roleMonitored.T == shared.CommunicationIIGORole {
+			if monitoringResult, ok := data[shared.MonitoringResult]; ok {
+				if monitoringResult.T == shared.CommunicationBool {
```
Comment (tobhil98) [inline internal/clients/team4/client.go]: Suggestion:
if monitoringResult, ok := data[shared.MonitoringResult]; 
    ok && monitoringResult.T == shared.CommunicationBool{

File: internal/clients/team4/client.go [hunk 3]
```diff
@@ -272,7 +285,47 @@ func (c *client) StartOfTurn() {
...
+					roleID := c.getRole(roleMonitored.IIGORoleData)
+
+					if monitoringResult.BooleanData {
+						// Monitored role was truthful
+						c.trustMatrix.ChangeClientTrust(roleID, c.internalParam.monitoringWeight*c.internalParam.monitoringResultChange) // config?
+					} else {
+						// Monitored role cheated
+						c.trustMatrix.ChangeClientTrust(roleID, -c.internalParam.monitoringWeight*c.internalParam.monitoringResultChange)
+					}
```
Comment (tobhil98) [inline internal/clients/team4/client.go]: Slightly experimental, but could you have something like a bool to +/-, so true evaluate to 1.0 and false evaluate to -1.0?

File: internal/clients/team4/client_test.go [hunk 4]
```diff
@@ -7,12 +7,68 @@ import (
...
 	winner := j.DecideNextPresident(shared.Team1)
 
 	if winner != shared.Team1 {
 		t.Errorf("Got wrong judge pointer. Winner is %v", winner)
 	}
+}
+
+func TestUpdateTrustFromHistory(t *testing.T) {
+	cases := []struct {
+		name         string
```
Comment (tobhil98) [inline internal/clients/team4/client_test.go]: You're not using this

File: internal/clients/team4/client_test.go [hunk 5]
```diff
@@ -7,12 +7,68 @@ import (
...
+
+			testServer := fakeServerHandle{clients: clients}
+			testClient := newClientInternal(id, t)
+			testClient.Initialise(testServer)
+
+			testClient.savedHistory = &tc.savedHistory
+			testClient.updateTrustFromSavedHistory()
+
+		})
+	}
```
Comment (tobhil98) [inline internal/clients/team4/client_test.go]: Are you just checking runtime crashes?

File: internal/clients/team4/judge.go [hunk 6]
```diff
@@ -59,24 +72,23 @@ func (j *judge) saveHistoryInfo(iigoHistory *[]shared.Accountability, lieCounts
 	for client, pairs := range accountabilityMap {
 		clientInfo, ok := buildHistoryInfo(pairs)
 		if ok {
-			clientLied := (*lieCounts)[client]
-			clientInfo.Lied = clientLied
+			truthfulRatio := (*truthfulness)[client]
```
Comment (tobhil98) [inline internal/clients/team4/judge.go:75]: Maybe rename?

**Review by tobhil98 — CHANGES_REQUESTED** [review]
See comments

File: internal/clients/team4/client.go (hunk 2 above)
Comment (akowal3) [reply internal/clients/team4/client.go]: ah, clever

File: internal/clients/team4/client_test.go (hunk 5 above)
Comment (akowal3) [reply internal/clients/team4/client_test.go]: I had a couple of prints there

File: internal/clients/team4/client.go (hunk 3 above)
Comment (akowal3) [reply internal/clients/team4/client.go]: I think this is more readable

File: internal/clients/team4/client.go (hunk 1 above)
Comment (akowal3) [reply internal/clients/team4/client.go:29]: I think I took it from @MikeIsMike. ie, he wasn't initialising them

**Pull request merged**